R/FilterStateDatettime.R

```diff
@@ -170,18 +161,31 @@ DatetimeFilterState <- R6::R6Class( # nolint
     #' @description
     #' Returns reproducible condition call for current selection.
     #' For this class returned call looks like
-    #' `<varname> >= as.POSIXct(<min>, tz = <timezone>) & <varname> <= <max>, tz = <timezone>)`
+    #' `<varname> >= as.POSIXct(<min>) & <varname> <= <max>)`
     #' with optional `is.na(<varname>)`.
     get_call = function() {
```

Comment:
I'm getting warnings from the time zone here.

![image](https://user-images.githubusercontent.com/36482959/221657687-e301528c-4c09-43c7-9c56-605f7857a4f2.png)

From what I can gather `attr(as.POSIXlt(...), "tzone")` is different from `Sys.timezone()` which was used before. The attribute returns "PST" for me, the latter "America/Los_Angeles". When printing a POSIXlt, it prints with "PST". This makes sense because that's typically what is used. However, when trying to specify that as a timezone it gives warnings because it expects it in the "America/Los_Angeles" format.

It looks like you're checking the timezone of the data? Maybe you could get the time zone this way, via attribute. Then check if it's in `olsonNames()`. If it's in this list, use that. Otherwise use `Sys.timezone()`. I think this could resolve the immediate warnings.

More generally, and potentially beyond the scope of this issue, what if we check the timezone of the data on initialization and store that in the class as a reference. I'm not sure if we have rules/guidelines around handling time zones. But to me it makes sense to keep the time zone of the data if present.

Reply:
The thing is I am not checking timezones anywhere and I am not getting the warnings you get. This looks like an automatic tz validation step and it is odd that it fails as PST should be a valid name. It certainly is on my machine (Ubuntu 20.04 R 4.2.2) although @BLAZEWIM had some issues with it.

Here is what a comparison returns:
```
> all.equal(
    as.POSIXct("2020-02-20 12:00:00", tz = "PST"), 
    as.POSIXct("2020-02-20 12:00:00", tz = "America/Los Angeles")
  )
[1] "'tzone' attributes are inconsistent ('PST' and 'America/Los Angeles')"
```

The reason I did NOT use Sys.timezone() is because of `airDatePicker` assuming you are in UTC, so results will vary depending on your local tz. For all tests, examples, etc. I set (or attempted to set) tz manually.